timber/docs
# 2.x – Use Eleventy to build documentation

This pull request is for the update to the new documentation generator using [Eleventy](https://www.11ty.dev/).

I’ve figured, that while [Hugo](https://gohugo.io/) is great for building static sites, it’s quite hard to work on new custom features. I was looking for other static site generators and first tried Gatsby, but then came back to [Eleventy](https://www.11ty.dev/) for its simplicity and ease of adding custom features.

So what’s better with this update?

- **Nunjucks** – We can use [Nunjucks](https://mozilla.github.io/nunjucks/) to build templates, which has the same syntax as Twig.
- **A better navigation** – I’ve built a custom script to build the navigation tree from the folder structure and the files in these folders. That way, we need to pass fewer meta data (front matter) to build the navigation.  
Also, the navigation is expanded all the time for a better overview over what’s in the different sections.
- **Updated styles** – I updated the colors a little and improved spacing between elements a bit. Not many people will notice the difference, I assume. I updated the code examples with Timber’s color scheme. We’ll see if that’s good enough or if we’d rather just take GitHub’s code style.
- **Sprinkles of JavaScript** I introduced [Turbolinks](https://github.com/turbolinks/turbolinks) as a progressive enhancement to a) load the pages even faster and b) make the navigation sidebar not scroll back to the top when clicking a link. And there’s some vanilla JavaScript for the mobile navigation.

## Other

- **Search should still work.** I will have to make a small pull request to the [**timber.json**](https://github.com/algolia/docsearch-configs/blob/master/configs/timber.json) file to fix some small things.
- **Redirects work differently.** To make redirects work with Eleventy, there’s now a central [**redirects.njk**](https://github.com/timber/timber/blob/2.x-docs-11ty/docs/v2/redirects.njk) file.
- Anchor links will be added to the headings themselves. This will make it easier to find a link to a section.
- Version switching works a little better than the one I envisioned with Hugo.

## How to test

It’s probably best to check out this branch and check the README. It should work to only run `npm run build`.
